### ISSUE TITLE
CerberusQA - Glassfish4 + Java 1.8 not able to run tests

### ISSUE THREAD

**Issue by mbessa** [cerberustesting/cerberus-source#1217]
Hello,

I think we have a regression, we aren't able to run tests in 192.168.135.30/CerberusQA, this is the server that has the requirements specified in https://github.com/cerberustesting/cerberus-source/blob/master/INSTALL
GlassFish Server Open Source Edition  4.1.1
java version "1.8.0_92"
10.1.13-MariaDB

When launch tests for THESEUS (Test=Performance+Monitor&TestCase=5014A&Country=IT&Environment=PREPROD)
Browser opens in VM
![image](https://cloud.githubusercontent.com/assets/5825695/23656890/71e8070c-0333-11e7-91d7-cbbd1b874c74.png)

Cerberus show's the following error:
![image](https://cloud.githubusercontent.com/assets/5825695/23656905/85b1ecc6-0333-11e7-8e67-f97aff532d5c.png)

In Glassfish and Selenium, similar errors are displayed.

If the launch a VCCRM test.
Nothing happens in VM and in Cerberus Internal Server Error is displayed.
![image](https://cloud.githubusercontent.com/assets/5825695/23657002/f9a6964a-0333-11e7-9aa2-407f8929de07.png)

Additional info:
The same tests launched in 197/CerberusQA, both work.
With previous Cerberus versions, we were able to run tests in this server.

Cerberus Version : 1.1.11-SNAPSHOT-1107

**Comment by bcivel** [comment]
Yes you can : http://192.168.135.30/CerberusQA/ExecutionDetail2.jsp?executionId=39917734

**Comment by cjmachado** [comment]
Hi,

That execution was run with firefox 24.5, according to the robot settings shown below:

![robot](https://cloud.githubusercontent.com/assets/7592715/23659543/58e4e350-033e-11e7-85f0-28e217ed0572.png)

That version of firefox was replaced by 31.6 more than a year ago. Also, from Cerberus it's not possible to find out which version of selenium server was used to run this test. Could you please specify the version?

**Comment by abourdon** [comment]
Hi @cjmachado,

So if I understand well, it is not a bug related to Glassfish 4 & Java 8, but more related to the compatible versions to use between Selenium & Firefox?

If yes, could you please edit this issue's title to be more explicit and reformulate your question?

Regards,
Aurélien

**Comment by cjmachado** [comment]
Hi @abourdon,

In fact, it might be related to all 4 pieces of software: Glassfish, Java, Selenium and Firefox but we don't know for sure. But one thing is certain: we should not be testing with that version of firefox (24.5). 

As @mbessa mentioned (the issue requester), the test case runs OK using the following combination of software: Glassfish 3.1.2.2, Java 7u80, Selenium 2.45 and Firefox 45.4.  
Furthermore, the test case fails to run with the following combination: Glassfish 4.1.1, Java 8u92, Selenium 2.45 and Firefox 45.4.

According to @bcivel execution, it seems to run OK with the following combination of sofware: Glassfish 4.1.1, Java 8u92, Selenium ??? and Firefox 24.5.

I hope I made myself clear.

**Comment by abourdon** [comment]
@cjmachado, there is a strong dependency between Selenium & browsers versions. Please take a look at http://www.seleniumhq.org/about/platforms.jsp

**Comment by cjmachado** [comment]
@abourdon, I'm perfectly aware of that. I know the pain it is to find a good combination between Firefox and Selenium. 
We are using Firefox 45.4 / Selenium 2.45 and it works fine for the production environment. It's not possible to run test cases from a Glassfish 4 server using the same versions (same Selenium and Firefox versions).

This isn't urgent at the moment, because we don't use Glassfish 4 yet. But please be warned that it's not working on that version of the application server.

Thanks and regards,

Carlos Machado

**Comment by mbessa** [comment]
Hello,

Is there any news on this subject? Did you solve this problem?
Or if you can't find any issue, can you provide us the correct combination of software versions for Cerberus, Hub + nodes to work properly?
Glassfish 4.1.1 ?
Java 8?? u92 ?
Firefox ESR 45.4 ? 52x ???
Selenium 2x? 3x ?
Any additional software?

We have some machines in standby, waiting to be tested/used due to this issue. It's important if you could provide some support on these matter.
Thanks!
Marco

**Comment by bcivel** [comment]
Hello @mbessa,

In fact, you have no strong dependency between cerberus server and the selenium server.

In the installation procedure (`https://github.com/cerberustesting/cerberus-source/blob/master/INSTALL`) , we advice Java, Glassfish and MySQL version for the cerberus server. For the selenium server, you can use the selenium version that you want, depending on your browser version.

For example, here is an execution with the configuration that fail on your local instance (gf 4.1.1, java 8, FF31, Selenium 2.52) `http://demo.cerberus-testing.org:28081/Cerberus/ExecutionDetail2.jsp?executionId=1759`

It may be related to your Cerberus installation, or some integration issue. I close that issue  and provide support if needed through private email.

Thanks
Benoit

**Comment by mbessa** [comment]
Hello,

Your test/execution don't match what is specified in documentation, FF version is way to old and Selenium is newer.

https://github.com/cerberustesting/cerberus-source/blob/master/INSTALL
> **MySQL 5.6.xx
Java 1.8
Glassfish Server v4.1.1**
......
Here is the configuration we could see that provide the best stability :
**Firefox 45.4.0ESR and Selenium 2.45**
Other earlier configuration that should provide good result are :
Firefox 52.0ESR and selenium 3.3.1 (with Gecko driver 0.15). <

With the configuration highlighted in bold isn't possible to execute any test. 
Note: Firefox 52 ESR still not stable and therefore not available in repo.
![image](https://cloud.githubusercontent.com/assets/5825695/26579427/b90850ee-452b-11e7-9542-c0e933003b94.png)

Can you please re-open this issue until is solved or update documentation accordingly?

Thanks,
MB

**Comment by vertigo17** [comment]
Marco, documentation is correct.
We currently use in production firefox 45.4.0 LINUX with Selenium 2.45 and it works fine with all the web campains with no significant issue.
In our view (we may be wrong) there should be no link between the selenium version and firefox version use and the cerberus reco on mysql/glassfish/java.

What happen here is that we managed to get a configuration that works on an installation and another that don't (in QA). That must be linked to some specific installation or configuration that is not linked with Cerberus in itself.
Let's keep this closed and solve that on a different internal thread.

We may of course adjust the documentation once we understand why it does not work in QA.

**Comment by mbessa** [comment]
Sorry but you are wrong... In production there is Glassfish 3...

**Comment by vertigo17** [comment]
Before answering, please read the comments from Benoit and I:

_In our view (we may be wrong) there should be no link between the selenium version and firefox version use and the cerberus reco on mysql/glassfish/java._

**Comment by mbessa** [comment]
Maybe i wasn't clear before, this issue was opened because we had Cerberus working in QA for a while until some point in time (think about the time Cerberus internal selenium was updated). Server configurations hadn't been changed (Java, Mysql/mariadb, Glassfish).

The things that were changed since work/not work were Cerberus version, Cerberus internal selenium version. At the nodes, firefox version, selenium version. 
The combination you specify don't even start the tests, and so far, none of the examples you had provided have the same as in the documentation.

Is possible to execute a test with MySQL 5.6.xx, Java 1.8, Glassfish Server v4.1.1, Firefox 45.4.0ESR and Selenium 2.45 ?
I know that with different selenium/firefox configurations it works, let me rephrase, it start the test, but some of these combinations don't work properly. All I'm asking is for you to provide the correct combination that is working configuration for Glassfish 4, since the one that is recommended in the documentation as far I can tell, isn't working.

Thanks!
Marco
